// _nuxtjs/mdc/dist/runtime/components/prose/ProseImg.vue3.mjs
import { defineComponent, computed, createVNode, resolveDynamicComponent, unref, mergeProps } from "vue";
import { ssrRenderVNode } from "vue/server-renderer";
import { withLeadingSlash, withTrailingSlash, joinURL } from "/Users/emanuele/Workspace/colonnaemanuele.github.io/node_modules/ufo/dist/index.mjs";
import ImageComponent from "../../../../../../../.nuxt/mdc-image-component.mjs";
import { useRuntimeConfig } from "../../../../../../nuxt/dist/app/nuxt.mjs";
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "ProseImg",
  __ssrInlineRender: true,
  props: {
    src: {
      type: String,
      default: ""
    },
    alt: {
      type: String,
      default: ""
    },
    width: {
      type: [String, Number],
      default: void 0
    },
    height: {
      type: [String, Number],
      default: void 0
    }
  },
  setup(__props) {
    const props = __props;
    const refinedSrc = computed(() => {
      var _a;
      if (((_a = props.src) == null ? void 0 : _a.startsWith("/")) && !props.src.startsWith("//")) {
        const _base = withLeadingSlash(withTrailingSlash(useRuntimeConfig().app.baseURL));
        if (_base !== "/" && !props.src.startsWith(_base)) {
          return joinURL(_base, props.src);
        }
      }
      return props.src;
    });
    return (_ctx, _push, _parent, _attrs) => {
      ssrRenderVNode(_push, createVNode(resolveDynamicComponent(unref(ImageComponent)), mergeProps({
        src: unref(refinedSrc),
        alt: props.alt,
        width: props.width,
        height: props.height
      }, _attrs), null), _parent);
    };
  }
});
export {
  _sfc_main as default
};
